### PR TITLE
Vectorization cleanup

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Currently, vectorization analysis is done by first calling `getInnerDimVectorizableWidth` to compute teh vectorization factor of the innermost dimension, then expand that to contiguous merge of `getMaxVectorizableWidth`. However, these two functions are very similar, almost a copy-paste of each other. Therefore, it makes no sense to use `getInnerDimVectorizableWidth` then `getMaxVectorizableWidth`, we should remove `getInnerDimVectorizableWidth` and go directly to `getMaxVectorizableWidth`.

This PR removes `getInnerDimVectorizableWidth`, anded a boolean option `contig_merge` to `getMaxVectorizableWidth` to reproduce the behavior of `getInnerDimVectorizableWidth`. The `contig_merge=false` is only used in the transpose scheduler, where the which dimensions are contiguously merge to which ID is more complicated and is not considered in heuristics.

Unfortunately, because the variable name change, I can not diff the kernel to check if they match exactly or not, but I do briefly skimmed through the diff by eye, and I don't find any change in the vectorization factor.

The V100 failure is OOM.

### CODE REVIEW

File: csrc/scheduler/vectorize_helper.cpp [hunk 1]
```diff
@@ -1201,13 +1201,29 @@ int64_t getVectorizationSize(
   return vectorize_size;
 }
 
-size_t getExpandedVectorization(
```
Comment (zasdfgbnm) [inline csrc/scheduler/vectorize_helper.cpp:1204]: A big portion of `getVectorizationFactor` has been removed, so I just cut-paste the function body of this function into `getVectorizationFactor` and remove this function.

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

File: csrc/scheduler/vectorize_helper.cpp [hunk 2]
```diff
@@ -1157,6 +1157,11 @@ int64_t getVectorizationSize(
     auto denominator = denominator_optional->as<int64_t>();
     auto extent = extent_optional->as<int64_t>();
 
+    // TODO: we should clean this up with expr simplifier
+    auto gcd = std::gcd(numerator, denominator);
+    numerator = numerator / gcd;
+    denominator = denominator / gcd;
```
[the remark this answers is not in the document]
Comment (naoyam) [reply csrc/scheduler/vectorize_helper.cpp:1163]: Seems to make sense. Did you actually find a case where this change matters?

**Review by naoyam — APPROVED** [review]
The changes make sense to me. Have you found any change in actual vectorization?

File: csrc/scheduler/vectorize_helper.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zasdfgbnm) [reply csrc/scheduler/vectorize_helper.cpp:1163]: Yes, `FusionReshapeVectorize_CUDA` would fail without this

**Comment by zasdfgbnm** [comment]
> The changes make sense to me. Have you found any change in actual vectorization?

No, I didn't find any. But I didn't check programmatically. I just used my eye to inspect the output of
```
difft --language cpp ../Fuser3/old-kernels/ new-kernels/ | grep load
```
and didn't find any.

File: csrc/scheduler/vectorize_helper.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zasdfgbnm) [reply csrc/scheduler/vectorize_helper.cpp:1163]: I didn't look into this, but my guess is, in the past, `getInnerDimVectorizableWidth` already returns 4, so even if `getMaxVectorizableWidth` is broken, we still get vectorize 4. But now, we don't use `getInnerDimVectorizableWidth` any more, so we are more sensitive to bugs in `getMaxVectorizableWidth`.

[the remark this answers is not in the document]
Comment (naoyam) [reply csrc/scheduler/vectorize_helper.cpp:1163]: OK, makes sense.

**Pull request merged**